### PR TITLE
vscode signing

### PR DESCRIPTION
**Pull request by danieljurek**
**DO NOT MERGE UNTIL 1es-redirect.yml is reverted** 

Uses `vsce` to prepare extension for signing and signs using ESRP. 

Signature verification example: 

```
> vsce verify-signature --packagePath E:\temp\vsix-sign\vsix-4\azure-dev-0.9.0-alpha.1.vsix --manifestPath E:\temp\vsix-sign\vsix-4\azure-dev-0.9.0-alpha.1.manifest --signaturePath E:\temp\vsix-sign\vsix-4\azure-dev-0.9.0-alpha.1.signature.p7s
Signature verification result: Success
Package file size:  722521
Signature archive file size:  9075
Manifest file size:  2644
Signature file size:  10359
Manifest file hash:  l6nQX6H2kYvVDTeM5Q+6+v3H6OLwvNgMKHTGQySTnWY=
Signature file hash:  1nMULsj/wyU9SApKOhgKu5d5O/jve4+oLOUaSQAgADg=
Package file hash:  rgG6yUjfu1vC4qJAhwRxA6zduhgx9oozVb68Q5HUHag=
ManifestPackageIntegrityPolicy succeeded.
SignatureIntegrityPolicy for primary signature succeeded.
SignatureValidityPolicy for primary signature succeeded.
SignatureIntegrityPolicy for primary timestamp signature succeeded.
SignatureValidityPolicy for primary timestamp signature succeeded.
X.509 certificate chain validation will use the current user's trust store for timestamping.
SignatureTrustPolicy for publisher primary signature timestamp signature succeeded.
X.509 certificate chain validation will use the current user's trust store for code signing.
SignatureTrustPolicy for publisher primary signature succeeded.
RepositoryCountersignaturePolicy for repository countersignature succeeded with NotPresent.
Primary signature type:  publisher primary signature
Primary signature status:  OK
Timestamp signature status:  OK
Repository countersignature status:  NotPresent
Repository countersignature timestamp status:  Unset
Exit code:  Success
```

### CODE REVIEW

**Review by weshaggard — APPROVED** [review]

File: eng/pipelines/templates/steps/publish-vscode.yml [hunk 1]
```diff
@@ -76,3 +82,15 @@ steps:
...
+    - task: AzureCLI@2
+      displayName: Publish (using vsce)
+      inputs:
+        azureSubscription: azure-sdk-vsmarketplace
+        scriptType: pscore
+        scriptLocation: inlineScript
+        workingDirectory: release
+        inlineScript: |
+          npm install -g @vscode/vsce
+          vsce publish --azure-credential
```
Comment (danieljurek) [inline eng/pipelines/templates/steps/publish-vscode.yml:96]: This portion cannot be tested without publishing the extension live. I'm trying to get confirmation that it'll work from someone with knowledge of the `vsce` tool.

Comment (danieljurek) [reply eng/pipelines/templates/steps/publish-vscode.yml:96]: To get this working: We need to specify the file paths for package, manifest, and signature.

**Comment by danieljurek** [comment]
Ready to merge. Get the build-tools changes in first.

**Pull request merged**